add additional hardware support for Hexiwear K64 module

davidzuhn opened the pull request:
This commit extends the support for the Hexiwear module to include more (but not yet all) of the internal components.   The red, green and blue LED pins have been added to the board/pinmux definition.  The haptic feedback motor has been defined (based on CONFIG_HAPTIC_FEEDBACK).  

The docking station interfaces have been defined (based on CONFiG_HEXIWEAR_DOCKING_STATION) and includes all of the pinmux settings for the Click board interfaces (GPIO, I2C, SPI).  

SPI 0-2 devices have been added (based on the schematic).   SPI0 has been tested with input and output devices on the Click interfaces.

Documentation has been extended for these new BSP settings, and links to more of the internal components have been added.   

Some sample programs have been added, as tests/examples for these hexiwear specific changes.   I will note that I'm new to Zephyr, so I do not claim these examples use best practices for Zephyr apps.   Suggestions *gladly* welcome.

davidzuhn committed:
hexiwear: added support for additional hardware

Device & pinmux setup for SPI 0-2
Docking station Click interface pin definitions
Additional links to onboard sensor datasheets
Pinmux setup for builtin device GPIO
Sample programs for these newly supported devices

Signed-off-by: david d zuhn <zoo@StateBeltRailway.org>

mnkp committed:
arch: sam3x: Use ASF library

This patch converts Atmel sam3x MCU series to use register
header files from Atmel Software Framework (ASF) library.
By using ASF different Atmel SAM MCU series can use common
device drivers.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

mnkp committed:
arch: atmel_sam: enhance gpio driver to support sam3x

This patch adds support for remaining Atmel SAM MCU series,
like sam3x to the common GPIO driver. After this update
full SAM MCU family should be supported by the driver.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

superna9999 committed:
arm: stm32l4: Add configuration and dts for STM32L496

Add configuration and dts for the STM32L496 SoC STM32L4 variant.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

superna9999 committed:
boards: arm: Add support for STM32L496G Discovery board

Add configuration, pinmux, dts and documentation for the STM32L496G
Discovery board based on the STM32L496AG SoC.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

superna9999 committed:
ext: stm32cube: update stm32l4xx cube version

Update Cube version for STM32L4XX family
from version: V1.6.0
to version: V1.8.0

Add support for following soc:
stm32l496xx
stm32l4a6xx

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

superna9999 committed:
arm: Add dts for STM32F407

Add missing dts for STM32F407.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

superna9999 committed:
arm: Add support for STM32F4DISCOVERY Board

Add configuration, documentation, flash script, pinmux and dts for the
STM32F4DISCOVERY board.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

galak committed:
arm: dts: cleanup HAS_DTS from defconfig

Since we now select HAS_DTS from the cortex-m Kconfig we don't need it
in 96b_carbon_nrf51_defconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
serial: dts: remove !HAS_DTS related Kconfig

As we now only support DTS for ARM based SoCs we can remove any
associated !HAS_DTS bits in Kconfig.  The Nordic NRF5 serial driver and
the ARM CMSDK APB serial drivers had Kconfig bits related to !HAS_DTS
builds.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

fallrisk committed:
arch: arm: Convert Atmel SAM4S series MCU to use ASF

Signed-off-by: Justin Watson <jwatson5@gmail.com>

galak committed:
dts/extract_dts_includes.py: Quote strings values

If the value of the property is a string quote it on output.  This lets
us use things like label properties that will have values of strings.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: stm32: dts: use label to generate STM32 uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the STM32 UART out of the device tree
instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: Fixup sanity check list of boards

The stm32f4discovery was incorrectly named and should have been
stm32f4_disco.  Also added 96b_carbon_nrf51 that was missing from the
list.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: Add label property to uart.yaml

Add optional label property to be generated as part of the UART yaml
spec.  This lets us use the label string in a device tree to generate
the device name (for example CONFIG_UART_STM32_PORT_1_NAME).

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: nrf: dts: use label to generate NRF uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the NRF UART out of the device tree
instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

MaureenHelm committed:
doc: Update NXP board docs for pyOCD and Segger J-Link

Now that NXP boards support 'make flash' and 'make debug' via pyOCD and
Segger J-Link, update all the NXP board docs accordingly. Adds a new
generalized OpenSDA document so certain details don't have to be
repeated for every OpenSDA board.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

galak committed:
arm: nxp_kinetis: dts: use label to generate NXP Kinetis uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the NXP Kinetis UART out of the device tree
instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: cmsdk_apb: dts: use label to generate ARM CMSDK APB uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the ARM CMSDK APB UART out of the device
tree instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: ti: cc32xx: dts: use label to generate TI CC32xx uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the TI CC32xx UART out of the device
tree instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: ti: stellaris: dts: use label to generate TI Stellaris uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the TI Stellaris UART out of the device
tree instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: atmel: sam3: dts: use label to generate Atmel SAM3 uart name

Now that we can utilize label in the device tree we can convert to
getting the device name for the Atmel SAM3 UART out of the device
tree instead of from Kconfig.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak reviewed (changes requested):
Can you rebase on the 'arm' branch, and cleanup the commits so they are cleaner and more logically grouped?

superna9999 committed:
stm32f4: Add support for STM32F469XI

Add configuration, dtsi and memory configuration fixup for the
STM32F469XI High Performance SoC.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

superna9999 committed:
boards: arm: Add support for STM32F469I-DISCO

Add board configuration, dts and pinmux based on the stm32f4_disco
board.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>

erwango committed:
dts: stm32: rename uart generic node name to serial

In stm32 dtsi linux files, uart generic node name is serial.
Rename uart node names of stm32 dtsi files from uart@ to serial@

Change-Id: Iac5cbf7955f23cee520bc1790b0f324a17bfcf9e
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak committed:
arm: dts: Add zephyr,console to missing device trees

The chosen property that sets which serial port is being used for the
console device wasn't set.  Add the property "zephyr,console" to the
Arduino Due, CC3200-LaunchXL, CC3220SF-LaunchXL, and QEMU Cortex M3
boards.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

erwango committed:
dts: stm32: uart nodes were missing yaml description

This commit fixes uart only nodes situation for stm32 devices.
st,stm32-uart yaml description is added to enable compilation

Change-Id: Iea78693bdfb90fbb09612b75685ed7ca0ccca6d6
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
board: stm32f469i_disco: fixup to compile uart

Fix copy/paste issue that block compilation on stm32f469i_disco

Change-Id: If8ab1087ad67274c67c91edf47980d333bf612c1
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: stm32f4: rework family tree

This commit rework stm32f4 series dtsi files hierarchy.
stm32f4.dtsi was used as maximum common set of IP while it should
be considered as the minimum common subset.
Then, following on stm32f4 series hierarchy and inheritance rules,
stm32f4xxx.dtsi files are reworked to include the "parent" soc dtsi
file and then add own IPs to each SoC.

Change-Id: I394278c84a8ea38921f9f143f4fc52ef1c645d05
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak committed:
arm: mps2_an385: remove setting of baud rate in defconfig

We now get the baud rate for the serial port from the device tree so we
don't need to be setting it in the defconfig.

The baud rate changed from 38400 to 115200 when we introduced the device
tree for MPS2.  The default baud rate was based on matching mbed, but we
use 115200 to match the standard default we have for most boards now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
arm: board: Cleanup stm32f4_disco & stm32l496g_disco defconfig

Removed CONFIG_HAS_DTS from the stm32f4_disco_defconfig &
stm32l496g_disco_defconfig files as its automatically selected for all
ARM SoCs at this point.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

dbkinder commented on boards/arm/hexiwear_k64/doc/hexiwear_k64.rst:
```diff
@@ -95,39 +98,62 @@ Connections and IOs
+also defined.   Each interface (numbered 1-3) defines the following
+
+   .. code-block:: console
+      CLICKn_AN_{NAME, PIN}
```
needs a blank line after the .. code-block line and before the first line of the code block contents (otherwise it complains about parameters to the codeblock tag)

dbkinder commented on samples/boards/hexiwear/4x4-buttons/README.rst:
```diff
@@ -0,0 +1,38 @@
+.. _hello_world:
```
document labels must be unique across the entire documentation set.  I'm wondering if you made a copy of the hello_world sample to start writing the 4x4-buttons sample and then didn't update the content?

dbkinder commented on samples/boards/hexiwear/blinky/README.rst:
```diff
@@ -0,0 +1,31 @@
+.. _blinky-sample:
```
document labels must be unique across the entire documentation set so give this a unique name (maybe hexiwear-blinky-sample)? . (There's already a blinky-sample in samples/basic/blinky/README.rst)

dbkinder commented on samples/boards/hexiwear/haptic/README.rst:
```diff
@@ -0,0 +1,31 @@
+.. _blinky-sample:
```
looks like another case of copying a documentation example and not updating it for the "haptic" sample

dbkinder commented on samples/boards/hexiwear/rotary-leds/README.rst:
```diff
@@ -0,0 +1,52 @@
+.. _hello_world:
```
needs a unique name, maybe "rotary-click-board"?

dbkinder commented on samples/boards/hexiwear/dock-blinky/README.rst:
```diff
@@ -0,0 +1,33 @@
+.. _blinky-sample:
```
give this a unique name, maybe "docking-station-blinky-sample"?

dbkinder reviewed (changes requested):
thanks for you first contribution! docs need unique labels across the whole doc set, looks like docs were copied to get started but then not updated for the new sample, need to add a samples index doc for hexiware as was done for other boards (see samples/boards/microbit/microbit.rst for example)

davidzuhn commented:
I'm withdrawing this pull request, and will soon create a new one rebased against the origin/arm branch.   I've cleaned up the commit history, and fixed up the doc/RST issues.
